commit e4e4e918ab9ee0f0509c416903b8f2b42cb5052d
Author: Josh Wilsdon <jwilsdon@joyent.com>
Date:   2019-04-15T14:51:22-07:00 (6 months ago)
    
    TRITON-1406 cn-agent should include newline when updating config
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/backends/smartos/common.js b/lib/backends/smartos/common.js
index cb9687b..821ca15 100644
--- a/lib/backends/smartos/common.js
+++ b/lib/backends/smartos/common.js
@@ -61,6 +61,7 @@ function zoneList(name, callback) {
 }
 
 function modifyConfig(configPath, key, value, callback) {
+    var newConfig;
     var out = [];
     var found = false;
 
@@ -81,7 +82,9 @@ function modifyConfig(configPath, key, value, callback) {
             out.push('overprovision_ratio=\''+value+'\'');
         }
 
-        fs.writeFile(configPath, out.join('\n'), 'utf8', function (writeError) {
+        newConfig = out.join('\n') + '\n';
+
+        fs.writeFile(configPath, newConfig, 'utf8', function (writeError) {
             callback(writeError);
         });
     });
diff --git a/package.json b/package.json
index 0332b44..9fed610 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cn-agent",
   "description": "Triton Compute Node Agent",
-  "version": "2.12.1",
+  "version": "2.12.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
